vaultwarden/templates/deployment.yaml

```diff
@@ -126,6 +126,24 @@ spec:
               value: {{ $val | quote }}
             {{- end }}
             {{- end }}
+            {{- if .Values.vaultwarden.extraEnvFromSecretKey }}
+            {{- range $key, $ref := .Values.vaultwarden.extraEnvFromSecretKey }}
```

Comment:
The existing implementation for extra environment variables uses `vaultwarden.extraEnv`. I am fine with the suggestion to use one key for both config maps and secrets but I would design the API as follows (at least if we don't deprecate the existing extraEnv and replace it with a Kubernetes pod-like syntax):

```yaml
vaultwarden:
  extraEnvFrom:
    MY_VAR:
      secretKeyRef:
        name: my-secret
        key: my-secret-key
    MY_OTHER_VAR:
      configMapKeyRef:
        name: my-config-map
        key: my-config-map-key
```

Looking forward to your reply 😀

Reply:
I see, what I had in mind was something like:

values.yaml
```yaml
...
  extraEnv:
    ABC: abc
  extraEnvFromSecret: my-secret
  extraEnvFromConfigMap: my-config-map
```

deployment.yaml
```yaml
...
spec:
  containers:
    - name: vaultwarden
      env:
         [...]
        - name: ABC
          value: "abc"
      envFrom:
        - configMapRef:
          name: my-config-map
        - secretRef:
          name: my-secret
```
It makes the configuration very short for the chart user and won't require them to know K8s YAML other than to build their own Secret/ConfigMap, which kubectl can help. Because envFrom imports the whole resource, it decouples the env variables from the Helm chart. The user just need to update the resource, no need to change Helm chart values and update the deployment itself.

Reply:
This is for a different use case I'd say. In my case I use a secret I don't create myself but one created by a third party app. I'm sure there are others with similar use cases, too. So I can't control which keys are included in the secret. I can implement what you described, too, but it does not solve my problem. I need to mount some (but not all) keys of a secret and I need to mount them under environment variables names that differ from the key names in the secrets. Therefore, it's necessary for me to have the option to mount individual secret keys. As I said, if you wish I can add an option that allows for mounting whole secrets, too, but this would be yet an additional option.

Reply:
Okay, seems like you didn't have time to read my reply. I simply implemented your suggestion now in addition to what I need. So the users can decide themselves. Those who need to mount single keys of secrets (like me) can reference them now. Those who simply want to mount whole secrets can do this by referencing a secret by name. The same applies for config maps. Hope this is fine now.